hp_list/main.c

```diff
@@ -230,14 +230,13 @@ static bool __list_find(list_t *list,
     if (atomic_load(prev) != get_unmarked(curr))
         goto try_again;
     while (true) {
-        if (!get_unmarked_node(curr))
-            return false;
         next = (list_node_t *) atomic_load(&get_unmarked_node(curr)->next);
         (void) list_hp_protect_ptr(list->hp, HP_NEXT, get_unmarked(next));
+        /* If the relationship between prev / curr and curr / next is different
```

Comment:
I have no confidence that the comment is precise enough to describe why we should `goto try_agin`.  Any advice if I can improve the comment?